src/lib/ui/SupportedDevices/SupportedDevices.tsx

```diff
@@ -1,27 +1,87 @@
+import { Link } from '@mui/material'
 import { useQuery } from '@tanstack/react-query'
 import axios from 'axios'
+import { useCallback, useEffect, useState } from 'react'
 
 import { Button } from 'lib/ui/Button.js'
-import type { DeviceModel } from 'lib/ui/SupportedDevices/types.js'
+import SupportedDevicesFilterArea from 'lib/ui/SupportedDevices/SupportedDevicesFilterArea.js'
+import type { DeviceModel, Filters } from 'lib/ui/SupportedDevices/types.js'
 
 import SupportedDeviceRow from './SupportedDeviceRow.js'
 import SupportedDevicesHeader from './SupportedDevicesHeader.js'
 
-export function SupportedDevices() {
+const BASE_URL = 'https://devicedb.seam.co/api/device_models/list'
+
+export interface SupportedDevicesProps {
+  // If true, show the filter area and search bar
+  showFilterArea?: boolean
+}
+
+export function SupportedDevices({ showFilterArea }: SupportedDevicesProps) {
+  const [allDeviceModels, setAllDeviceModels] = useState<DeviceModel[]>([])
+  const [filterStr, setFilterStr] = useState('')
+  const [filters, setFilters] = useState<Filters>({
+    supportedOnly: false,
+    category: null,
+    brand: null,
+  })
+
+  const fetchDeviceModels = useCallback(async () => {
+    const queries = []
+
+    if (filterStr.trim() !== '') {
+      queries.push(`text_search=${encodeURIComponent(filterStr.trim())}`)
+    }
+
+    if (filters.supportedOnly) {
+      queries.push('support_level=live')
+    }
+
+    if (filters.category !== null) {
```

Comment:
Not currently used (UI is commented out) because all devices are currently `smartlocks`. Unused, but ready to be.

Reply:
#181